File: Zad2.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Wisielec
{
    class Player
    {
        public string Name { get; private set;}
        public Player(string name)
        {
            Name = name;
        }
    }

    class WordBank
    {
        private List<string> words = new List<string>
        {
            "prokrastynacja",
            "elektrokardiogram",
            "wyindywidualizowany",
            "Hiszpania",
            "wulkanizacja"
        };
        private Random random = new Random();

        public string GetRandomWord()
        {
            int index = random.Next(words.Count);
            return words[index];
        }
    }

    class Game
    {
        private Player player;
        private string secretWord;
        private HashSet<char> guessedLetters;
        private int maxErrors;
        private int errors;

        public Game(Player player, WordBank wordBank, int maxErrors = 6)
        {
            this.player = player;
            this.secretWord = wordBank.GetRandomWord().ToLower();
            this.guessedLetters = new HashSet<char>();
            this.maxErrors = maxErrors;
            this.errors = 0;
        }

        public string GetDisplayWord()
        {
            var display = secretWord.Select(c => guessedLetters.Contains(c) ? c : '_');
            return string.Join(" ", display);
        }
        public void GuessLetter(char letter)
        {
            letter = char.ToLower(letter);
            if (guessedLetters.Contains(letter))
            {
                Console.WriteLine("Już zgadłeś tę literę!");
                return;
            }
            guessedLetters.Add(letter);
            if (secretWord.Contains(letter))
            {
                Console.WriteLine($"Ta litera jest w słowie.");
            }
            else
            {
                errors++;
                Console.WriteLine($"Niestety, ta litera nie jest w słowie. Błąd {errors}/{maxErrors}");
            }
        }

        public bool IsWon() => secretWord.All(c => guessedLetters.Contains(c));
        public bool IsLost() => errors >= maxErrors;
        public void StartLoop()
        {
            Console.WriteLine($"Witaj {player.Name}! Zacznijmy grę w wisielca!");
            while (!IsWon() && !IsLost())
            {                Console.WriteLine($"Słowo: {GetDisplayWord()}");
                Console.Write("Zgadnij literę: ");
                
                string input = Console.ReadLine();
                if (string.IsNullOrEmpty(input) || input.Length != 1 || !char.IsLetter(input[0]))
                {
                    Console.WriteLine("Wprowadzić pojedynczą literę.");
                    continue;
                }
                GuessLetter(input[0]);
            }
            if (IsWon())
            {
                Console.WriteLine($"Gratulacje {player.Name}, wygrałeś! Słowo to: {secretWord.ToUpper()}");
            }
            else
            {
                Console.WriteLine($"Niestety {player.Name}, przegrałeś. Słowo to: {secretWord.ToUpper()}");
            }
        }
    }
    class Program
    {
        static void Main(string[] args)
        {
            Console.Write("Podaj swoje imię: ");
            string name = Console.ReadLine();
            Player player = new Player(name);
            WordBank wordBank = new WordBank();
            Game game = new Game(player, wordBank);
            game.StartLoop();
        }
    }
}

File: Zad2.py
import random

class Player:
    def __init__(self, name):
        self.name = name

class WordBank:
    def __init__(self):
        self.words = ["prokrastynacja", "python", "Hiszpania", "wulkanizacja", "elektrokardiogram"]

    def get_random_word(self) -> str:
        return random.choice(self.words)

class Game:
    def __init__(self, player: Player, word_bank: WordBank, max_errors: int = 6):
        self.player = player
        self.word_bank = word_bank
        self.max_errors = max_errors
        self.secret_word = word_bank.get_random_word().lower()
        self.guessed_letters = set()
        self.errors = 0
    
    def get_display_word(self) -> str:

        return " ".join([char if char in self.guessed_letters else "_" for char in self.secret_word])
    def guess_letter(self, letter: str) -> bool:
        letter = letter.lower()

        if letter in self.guessed_letters:
            print(f"Już zgadłeś literę '{letter}'.")
            return False
        
        if letter in self.secret_word:
            self.guessed_letters.add(letter)
            print(f"Dobra robota! Litera '{letter}' jest w słowie.")
            return True
        else:
            self.errors += 1
            print(f"Niestety, litera '{letter}' nie jest w słowie. Błąd {self.errors}/{self.max_errors}.")
            return False

    def is_won(self) -> bool:
        return all(char in self.guessed_letters for char in self.secret_word)
    
    def is_lost(self) -> bool:
        return self.errors >= self.max_errors
    
    def start_loop(self):
        print(f"Witaj, {self.player.name}! Zacznijmy grę w wisielca!")

        while not self.is_won() and not self.is_lost():
            print(f"Słowo: {self.get_display_word()}")
            letter = input("Zgadnij literę: ").strip()
        
            if len(letter) != 1 or not letter.isalpha():
                print("Proszę, wpisz pojedynczą literę.")
                continue

            self.guess_letter(letter)

        if self.is_won():
            print(f"Gratulacje, {self.player.name}! Wygrałeś! Słowo to '{self.secret_word.upper()}'.")
        else:
            print(f"Niestety, przegrałeś. Słowo to '{self.secret_word.upper()}'.")

if __name__ == "__main__":
    player_name = input("Podaj swoje imię: ")
    player = Player(player_name)
    word_bank = WordBank()
    game = Game(player, word_bank)
    game.start_loop()